shuffle/shuffle_test.go

```diff
@@ -64,3 +70,82 @@ func shuffleTest(suite Suite, k, N int) {
 		}
 	}
 }
+
+func sequenceShuffleTest(suite Suite, k, NQ, N int) {
+	rand := suite.RandomStream()
+
+	// Create a "server" private/public keypair
+	h := suite.Scalar().Pick(rand)
+	H := suite.Point().Mul(h, nil)
+
+	// Create a set of ephemeral "client" keypairs to shuffle
+	c := make([]kyber.Scalar, k)
+	C := make([]kyber.Point, k)
+
+	for i := 0; i < k; i++ {
+		c[i] = suite.Scalar().Pick(rand)
+		C[i] = suite.Point().Mul(c[i], nil)
+		//		fmt.Println(" "+C[i].String())
+	}
+
+	X := make([][]kyber.Point, NQ)
+	Y := make([][]kyber.Point, NQ)
+
+	// generate random sequences
+
+	for i := 0; i < NQ; i++ {
+		xs := make([]kyber.Point, k)
+		ys := make([]kyber.Point, k)
+
+		for i := 0; i < k; i++ {
+			xs[i] = suite.Point().Mul(suite.Scalar().Pick(suite.RandomStream()), nil)
+			ys[i] = suite.Point().Mul(suite.Scalar().Pick(suite.RandomStream()), nil)
+		}
+
+		X[i] = xs
+		Y[i] = ys
+	}
+
+	// ElGamal-encrypt all these keypairs with the "server" key
+	r := suite.Scalar() // temporary
+	for j := 0; j < NQ; j++ {
+		for i := 0; i < k; i++ {
+			r.Pick(rand)
+			X[j][i] = suite.Point().Mul(r, nil)
+			Y[j][i] = suite.Point().Mul(r, H) // ElGamal blinding factor
+			Y[j][i].Add(Y[j][i], C[i])        // Encrypted client public key
+		}
+	}
+
+	// Repeat only the actual shuffle portion for test purposes.
+	for i := 0; i < N; i++ {
+
+		// Do a key-shuffle
+		XX, YY, getProver := SequencesShuffle(suite, nil, H, X, Y, rand)
+
+		e := make([]kyber.Scalar, NQ)
+		for j := 0; j < NQ; j++ {
+			e[j] = suite.Scalar().Pick(suite.RandomStream())
+		}
+
+		prover, err := getProver(e)
+		if err != nil {
```

Comment:
Here and other tests for err: pass the `*testing.T` in this method, then do the 'normal' `require.NoError(t, err)`